srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java

```diff
@@ -0,0 +1,260 @@
+import org.l2x6.srcdeps.core.config.ScmRepository;
+import org.l2x6.srcdeps.core.shell.IoRedirects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link LocalRepositoryManager} able to build the requested artifacts from their sources.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class SrcdepsLocalRepositoryManager implements LocalRepositoryManager {
+
+    /**
+     * Encapsulates the loading of {@link Configuration} that has to be done lazily - i.e. at some point when
+     * {@link MavenSession} is available already.
+     */
+    private static class LazyConfiguration {
+        private final Provider<MavenSession> sessionProvider;
+
+        public LazyConfiguration(Provider<MavenSession> sessionProvider) {
+            super();
+            this.sessionProvider = sessionProvider;
+        }
+
+        private volatile Configuration configuration;
+        private final Object configurationLock = new Object();
+
+        /**
+         * Loads the {@link Configuration} lazily. If multiple threads call this method while {@link #configuration} is
+         * uninitialized, only the first one loads from the file system while others are waiting.
+         *
+         * @return a {@link Configuration}
+         */
+        public Configuration getConfiguration() {
+            synchronized (configurationLock) {
+                if (configuration == null) {
+                    File srcdepsYaml = locateSrcdepsYaml();
+                    log.debug("SrcdepsLocalRepositoryManager using {}", srcdepsYaml.getAbsolutePath());
+                    try (Reader r = new InputStreamReader(new FileInputStream(srcdepsYaml), "utf-8")) {
+                        configuration = new YamlConfigurationIo().read(r);
+                    } catch (IOException | ConfigurationException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+            return configuration;
+        }
+
+        /**
+         * Finds the {@link ScmRepository} associated with the given {@code artifact}. The association is given by the
+         * exact string match between the groupId of the {@code artifact} and one of the
+         * {@link ScmRepository#getSelectors() selectors} of {@link ScmRepository}
+         *
+         * @param artifact
+         * @return
+         */
+        public ScmRepository findScmRepo(Artifact artifact) {
+            final String groupId = artifact.getGroupId();
+            for (ScmRepository scmRepository : getConfiguration().getRepositories()) {
+                if (scmRepository.getSelectors().contains(groupId)) {
+                    return scmRepository;
+                }
+            }
+            throw new IllegalStateException(String
+                    .format("No srcdeps SCM repository configured in .mvn/srcdeps.yaml for groupId [%s]", groupId));
+        }
```

Comment:
Even if it doesn't, does it make sense to have it on Java 7? It's not like Srcdeps would require the projects consuming it to be under Java 8, is it? It's just a runtime requirement for build-time, not for production.